src/main/java/org/opensearch/security/auth/BackendRegistry.java

```diff
@@ -224,6 +224,7 @@ public boolean authenticate(final SecurityRequestChannel request) {
         if (adminDns.isAdminDN(sslPrincipal)) {
             // PKI authenticated REST call
             threadContext.putTransient(ConfigConstants.OPENDISTRO_SECURITY_USER, new User(sslPrincipal));
+            threadContext.putPersistent(ConfigConstants.OPENDISTRO_SECURITY_USER, new User(sslPrincipal));
```

Comment:
How is this change related to resource sharing?

Reply:
In long term we do intend to change this to a persistent header and remove transient completely.